sql/pipelines/src/test/scala/org/apache/spark/sql/pipelines/graph/MaterializeTablesSuite.scala

```diff
@@ -446,8 +446,9 @@ class MaterializeTablesSuite extends BaseCoreExecutionTest {
 
     val table2 = catalog.loadTable(identifier)
     assert(
-      table2.columns() sameElements CatalogV2Util
-        .structTypeToV2Columns(new StructType().add("y", IntegerType).add("x", BooleanType))
+      table2.columns().toSet == CatalogV2Util
```

Comment:
why do we need this change?

Reply:
The ordering of columns does not appear to be deterministic (at least across different catalog implementations). Is that unexpected?

Reply:
I think the issue is that, the hive catalog always puts partition columns at the end of the table schema. For example, `a, b, c PARTITIONED BY b` will end up with column order `a, c, b`.

Can we tweak this test to avoid this hive limitation?

Reply:
So it sounds like the order of the columns returned here is not defined in a catalog-agnostic way? Objections to using non-ordered comparison here in that case?

An alternative could be for `MaterializeTablesSuite` to have a method that the upcoming Hive-specific subclass overrides. Though not sure the added complexity is worth it given that column ordering is tangential to the main behaviors we're trying to test in this case.

Reply:
nice find.  Does the suite run against hive/non-hive?  For me, Im ok either way, maybe add a short comment in the test to explain.

Reply:
I think we should fix the Hive catalog to respect the user-specified column order. For now, I'm fine with ignoring some test cases temporarily in the upcoming hive suite, by overriding `def excluded` from `SparkFunSuite`, or add column-order methods and override it in the hive suite.

Reply:
@cloud-fan @sryza @gengliangwang  i made a patch here, does it look like what we want?

Reply:
https://github.com/apache/spark/pull/51342

Reply:
OK I've updated this PR to bring back the asserts on the ordering, and we can deal with the Hive tests in the PR that introduces them. @cloud-fan mind taking another look?

Reply:
> 
Hmm for some reason the original order is working now. Switched it back and now the tests are passing.

@sryza so may not need https://github.com/apache/spark/pull/51342 after all?  :)  We can close it then if its not needed.  At least we managed to make the underlying API better in https://github.com/apache/spark/pull/51373, that fixes a small issue (SPARK-52681).  By the way, it also doesnt prevent you to drop the column and replace them, can you give a try as well after this pr?